blockchain/forkid/forkid.go

```diff
@@ -0,0 +1,152 @@
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The Kaia library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the Kaia library. If not, see <http://www.gnu.org/licenses/>.
+
+package forkid
+
+import (
+	"encoding/binary"
+	"hash/crc32"
+	"math/big"
+	"reflect"
+	"strings"
+
+	"github.com/kaiachain/kaia/common"
+	"github.com/kaiachain/kaia/params"
+)
+
+// ID is a fork identifier as defined by EIP-2124.
+type ID struct {
+	Hash [4]byte // CRC32 checksum of the genesis block and passed fork block numbers
+	Next uint64  // Block number of the next upcoming fork, or 0 if no forks are known
+}
+
+// NewID calculates the Kaia fork ID from the chain config, genesis hash, head.
+func NewID(config *params.ChainConfig, genesis common.Hash, head uint64) ID {
+	// Calculate the starting checksum from the genesis hash
+	hash := crc32.ChecksumIEEE(genesis[:])
+
+	// Calculate the current fork checksum and the next fork block
+	var next uint64
+	for _, fork := range gatherForks(config) {
+		if fork <= head {
+			// Fork already passed, checksum the previous hash and the fork number
+			hash = checksumUpdate(hash, fork)
+			continue
+		}
+		next = fork
+		break
+	}
+	return ID{Hash: checksumToBytes(hash), Next: next}
+}
+
+// LatestForkCompatibleBlock returns the latest fork compatible block or genesis(0) if no forks are known.
+func LatestForkCompatibleBlock(config *params.ChainConfig, head *big.Int) *big.Int {
+	latestForkCompatibleBlock := common.Big0
+	for _, fork := range gatherForks(config) {
+		if new(big.Int).SetUint64(fork).Cmp(head) <= 0 {
+			latestForkCompatibleBlock = new(big.Int).SetUint64(fork)
+			continue
+		} else {
+			break
+		}
+	}
+	return latestForkCompatibleBlock
+}
+
+// NextForkCompatibleBlock returns the next fork compatible block or nil if no forks are known.
+func NextForkCompatibleBlock(config *params.ChainConfig, head *big.Int) *big.Int {
+	var nextForkCompatibleBlock *big.Int
+	for _, fork := range gatherForks(config) {
+		if new(big.Int).SetUint64(fork).Cmp(head) <= 0 {
+			continue
+		}
+		nextForkCompatibleBlock = new(big.Int).SetUint64(fork)
+		break
+	}
+	return nextForkCompatibleBlock
+}
+
+// LastForkCompatibleBlock returns the last fork compatible block or genesis(0) if no forks are known.
+func LastForkCompatibleBlock(config *params.ChainConfig) *big.Int {
+	forks := gatherForks(config)
+	if len(forks) == 0 {
+		return common.Big0
+	}
+	return new(big.Int).SetUint64(forks[len(forks)-1])
+}
+
+func BlobConfig(config *params.ChainConfig, head uint64) *params.BlobConfig {
+	return nil
+}
```

Comment:
I think this funciton is more appropriate to be in params/config.go. After 4844, we're going to declare OsakaBlobConfig in params/config.go. This function will return that. e.g.

```
func (c *ChainConfig) BlobConfig(num uint64) *BlobConfig
```

Reply:
I agree, in the future it will be implemented as follows. Thanks.
```
if c.IsOsakaForkEnabled(head) {
	return OsakaBlobScheduleConfig
}
return nil
```